compiler/src/dotty/tools/dotc/quoted/Toolbox.scala

```diff
@@ -24,8 +24,8 @@ object Toolbox {
     def run(expr: Expr[T]): T = expr match {
       case expr: LiftedExpr[T] =>
         expr.value
-      case expr: TreeExpr[Tree] @unchecked =>
-        new QuoteDriver().run(expr.pickled, runSettings)
+      case expr: TastyTreeExpr[Tree] @unchecked =>
+        throw new scala.quoted.QuoteError("Can't run and scala.quoted.Expr coming from a macro argument")
```

Comment:
The comment as is has a typo, so a fix can be "Cannot run a type scala.quoted.Expr" ... 

However that error is thrown at the call site of an inline def. The `f.run` is invoked inside the macro body, so maybe we can rephrase into something the following maybe:

> Cannot call an inline definition that runs an Expr that comes from a macro argument.